### DISCUSSION TITLE
While testing testfulfillRandomWordsPicksAWinnerResetsAndSendMoney got Evm revert.

### DISCUSSION THREAD

**Discussion by Parth-Sharma-tonchi** [Cyfrin/foundry-full-course-cu#699] [Q&A]
I am stuck with this error.
`0x8103B0A8A00be2DDC778e6e7eaa21791Cd364625::fulfillRandomWords(112516256707324845614270895982487698080257839176050866886067761179462970124557 [1.125e77], Raffle: [0xA8452Ec99ce0C64f20701dB7dD3abDb607c00496]) 
    │   └─ ← "EvmError: Revert"
    └─ ← "EvmError: Revert"

Test result: FAILED. 14 passed; 1 failed; 0 skipped; finished in 7.45s
Ran 1 test suites: 14 tests passed, 1 failed, 0 skipped (15 total tests)

Failing tests:
Encountered 1 failing test in test/unit/TestRaffle.t.sol:RaffleTest
[FAIL. Reason: EvmError: Revert] testfulfillRandonWordsPicksAWinnerResetsAndSendMoney() (gas: 307480)`

**I think this error is getting because of parameters of fulfillRandomWords function. In this function the first argument is uint256 and another is an address. I passed the uint256(requestId) and address(raffle) to the function but it shows a bytes32 as a parameter. Correct me if I am wrong.
and help me to get out of this error.**

Following is the code to testfulfillRandomWordsPicksAWinnerResetsAndSendMoney:

```
 `    function testfulfillRandomWordsPicksAWinnerResetsAndSendMoney() public RaffleEnterAndTimePassed{
        uint256 additionalEntrants = 5;
        uint256 startingIndex = 1;
        uint256 prizeMoney = entranceFee * (additionalEntrants+1);

        for(uint256 i = startingIndex; i<=additionalEntrants; ++i){
            address lotteryPlayer = address(uint160(i));
            hoax(lotteryPlayer, STARTING_PLAYER_BALANCE);
            raffle.EnterRaffle{value: entranceFee}();
        }
        uint256 previousTimeStamp = raffle.getLastTimeStamp();


        vm.recordLogs();
        raffle.performUpkeep(""); // emits requestId
        Vm.Log[] memory entries = vm.getRecordedLogs();
        bytes32 requestId = entries[1].topics[1];

        uint256 requestId_uint = uint256(requestId);
        console.log(requestId_uint);

        VRFCoordinatorV2Mock(vrfCoordinatorV2).fulfillRandomWords(uint256(requestId), address(raffle));

        assert(raffle.getRecentWinner() != address(0));
        assert(raffle.getRaffleState() == Raffle.RaffleState.OPEN);
        assert(raffle.getNumberOfPlayers() == 0);
        assert(previousTimeStamp < raffle.getLastTimeStamp());
        assert(raffle.getRecentWinner().balance == STARTING_PLAYER_BALANCE + prizeMoney - entranceFee);
    }`
```

Help me to deal with this Error.

**Comment by shikhar229169** [comment]
There is something unusual with the requestId being fetched from the events, can you provide the whole traces of error.

**Reply by shikhar229169** [reply]
Also, try passing 1 to the fulfillRandomWords function for subId parameter. Does it work?

**Comment by theirrationalone** [comment]
You're not emitting `RequestedRaffleWinner` event again for your this specific test. `RequestedRaffleWinner` event on raffle has data into raffle contract's log here you need to copy that same event paste into this test and emit again performUpkeep..... 

Like this... :

```solidity
vm.recordLogs();
// vm.expectEmit(true, false, false, false, address(raffle) ); // on here : cheat Activated 🤪
raffle.performUpkeep(""); // emits requestId
emit RequestedRaffleWinner(1); // on raffle main contract == same
Vm.Log[] memory entries = vm.getRecordedLogs();
bytes32 requestId = entries[0].topics[1];
```
